lib/ssh/src/ssh_connection.erl

```diff
@@ -946,17 +946,27 @@ handle_msg(#ssh_msg_channel_open{channel_type = "direct-tcpip",
                        connection_supervisor = ConnectionSup
                       } = C,
 	   server, _SSH) ->
+    HostName = binary_to_list(HostToConnect),
```

Comment:
can we avoid doing this conversion for scenario when fun is not provided?
and do it only when it must be done (for example case clause for fun variant ..., setting to `undefined` if not needed?)

Reply:
yes. i will change the PR to avoid `binary_to_list(HostToConnect)` when the fun is not provided.